File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] results = new int[2];

        for (int i = 0; i < nums.length; i++){
            int diff = target - nums[i];
            System.out.println(diff);
            Integer partner = map.get(diff);
            if (partner != null) {
                System.out.println(partner);
                System.out.println(i);
                results[0] = partner;
                results[1] = i;
                return results;
            } else {
                map.put(nums[i], i);
            }
        }


        return null;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1

        while l < r:
            calc = numbers[l] + numbers[r]

            if calc == target:
                return [l+1, r+1]

            if calc > target:
                r -= 1
            
            if calc < target:
                l += 1
            
            
            